### ISSUE TITLE
Hemtt does not binarize materials for .p3d objects

### ISSUE THREAD

**Issue by AdlerSalbei** [BrettMayson/HEMTT#321]
**HEMTT Version:** `0.7.6`
**Project:**https://github.com/gruppe-adler/grad_trenches

**Description:**

Using hemtt, it does not binarize the materials for .p3d's correctly.

**Steps to reproduce:**

1. Binarize the above project.
2. See ingame that the trench objects are all smooth.

**Comment by ArmedVeteran** [comment]
I most likely have a similar error. I've added CBA to my mod (with a friend, who has experience with both CBA and packing) and packed it with HEMTT. Now my vests load only texture, but rvmats are somehow ignored. Idk what's the problem, because there are no errors, no missing files (neither prompts nor in RPT files). I'm pretty sure it's a packing error.

**Comment by jokoho48** [comment]
that I think actually has to do with how hemtt handles the path and how Arma expects those paths to be. also Rvmats are actually binarised into the p3d and uses the path that is given in the model. exception here is when you use hiddenSelections and normally RVMats don't need to deliver in the PBO and are only needed for the binarizing process.

**Comment by bux** [comment]
I'm on hemtt 0.7.3 and rvmats are being packed properly. All _as _smdi and _nohq files are being loaded along with the _co when viewing a model in game.

**Comment by ArmedVeteran** [comment]
@jokoho48 As for the first solution: I'm using those in the config, doesn't change anything really. Textures load, but rvmats don't (I'm not really sure anymore - is it supposed to be hiddenSelectionsMaterials or hiddenSelectionMaterials?). As for the prebinarization of the P3D - I'm pretty sure that HEMTT throws errors if it has any binarized files to deal with (since it binarizes them itself), so that wouldn't work. The last solution with the temp folder I don't really understand :/

**Comment by ArmedVeteran** [comment]
I packed with Mikero's and it works great :D

**Comment by AdlerSalbei** [comment]
The Issue seems to be, that the path that hemtt needs is different to the one binarize needs.
@Dahlgren fixed it for me with this script: https://github.com/gruppe-adler/grad_trenches/blob/master/build-hemtt.bat

**Comment by bux** [comment]
Using ACE's way of setting up the dev environment would be the way to go
https://ace3mod.com/wiki/development/setting-up-the-development-environment.html#41-initial-setup

**Comment by jokoho48** [comment]
@bux but isnt that the thing what hemtt wants to remove from the dev setup. to have a fully setup P Drive?

**Comment by bux** [comment]
I'm talking about the symlinking

**Comment by jonpas** [comment]
No bux, HEMTT already does not need a P-drive. Other symlink is to Arma 3 directory so you can write shorter mod lines (startup parameter).

**Comment by bux** [comment]
Didn't say it NEEDs a fully set-up P drive :P

**Comment by jonpas** [comment]
```toml
check = [
    "!binarizer_link"
]

# Binarizer file search compatibility
[scripts.binarizer_link]
steps_windows = [
    "if not exist x\\tacs mkdir x\\tacs",
    "if not exist x\\tacs\\addons mklink /j x\\tacs\\addons addons"
]
steps_linux = [
    # Preparation for Windows Binarizer (run once on Linux for network share compatibility)
    "mkdir -p x/tacs",
    "ln -sr addons x/tacs/addons"
]
show_output = true
```
You don't need the `steps_linux`, I just have it because I build off a network shared drive and `mklink` doesn't work with those.

**Comment by BrettMayson** [comment]
This should be fixed with 1.0 if I am not mistaken
